### PR TITLE
Equipment Caching: Rework

### LINKED ISSUE #1091: [Hotreload] Hotreload breaks shops
Shops with standard items (not with manually selected items) are broken after hotreload. This is because the functions has to be called after a hotreload: https://github.com/TTT-2/TTT2/blob/master/gamemodes/terrortown/gamemode/client/cl_equip.lua#L1117-L1130

### LINKED ISSUE #1403: unarmed errors because it's ejected
it has an .Icon but it never gets converted to .iconMaterial

because shopeditor tries to eject it

### PR DESCRIPTION
**Pull request by TimGoll**
Fixes #1403 
Partly fixes some of #1091 (it does not fix this issue, it fixes a part of this issue, but after hotreload weapons are still missing (items are there) and everything is not buyable)

It looks like lots of changes, but I only moved things around. It is mostly the old code with renamed variables split into three functions. Notable changes:

- caching is now independant of eject and other changes, meaning that also `weapon_ttt_unarmed` has a cached icon
- caching now happens on every load / reload of the game instead of the first call this improved the first opening time of the shop for me by a lot
- it fixes the issue that the cache is not rebuilt on hoteload until the shop is reopened, improving some things mentioned in in 1091

### CODE REVIEW

**Comment by TimGoll** [comment]
This should fix the issue @EntranceJew was experiencing. Please confirm

**Review by saibotk — APPROVED** [review]
Very nice!

**Comment by TimGoll** [comment]
> Looking good so far. Gonna check in detail at home.
> 
> Im only wondering if this is shopeditor only or if this actually should be done for shop or other parts as well. Because then I would move this to another namespace like shop or equipment related.

The shop UI already uses this, the shop backend has its own handling. Maybe one can unify this, but probably not in this PR

File: gamemodes/terrortown/gamemode/client/cl_equip.lua [hunk 1]
```diff
@@ -1195,40 +1193,6 @@ function GM:OnContextMenuOpen()
     end
 end
 
----
--- Caches the material or model
--- @param table item
--- @realm client
-function TTT2CacheEquipMaterials(item)
```
Comment (ZenBre4ker) [inline gamemodes/terrortown/gamemode/client/cl_equip.lua:1202]: This function is used in sh_main after a hotreload of a weapon you changed.

**Review by ZenBre4ker — CHANGES_REQUESTED** [review]

**Review by ZenBre4ker — APPROVED** [review]
Fixed the only error I saw. Otherwise looking good and works for me.

**Pull request merged**